src/allotropy/parsers/biorad_bioplex/biorad_bioplex_structure.py

```diff
@@ -0,0 +1,268 @@
+                sample_identifier = get_val_from_xml(
+                    child_sample_type, 1, "Sample Identifier"
+                )
+                if child_sample_type[2].tag == DILUTION:
+                    sample_dilution = get_val_from_xml(child_sample_type, 2, "Dilution")
+                    for child in child_sample_type:
+                        if child.tag == MEMBER_WELLS:
+                            for member_well in child:
+                                sample_document = (
+                                    SampleDocumentAggregate._generate_sample_document(
+                                        sample_documents=sample_documents,
+                                        well_xml=member_well,
+                                        sample_id=sample_identifier,
+                                        sample_dilution=float(sample_dilution),
+                                        sample_type=sample_type,
+                                        sample_description=sample_description,
+                                    )
+                                )
+                                sample_documents.samples.append(sample_document)
+
+        return sample_documents
+
+    @staticmethod
+    def _generate_sample_document(
+        sample_documents: SampleDocumentAggregate,
+        well_xml: ElementTree.Element,
+        sample_id: str,
+        sample_dilution: float,
+        sample_type: str,
+        sample_description: str,
+    ) -> SampleDocument:
+        well_name = get_well_name(well_xml.attrib)
+        mappings = WellAnalyteMapping(well_name=well_name, analytes=[])
+        for analyte in well_xml[0]:
+            # Create the analyte.
+            new_analyte = AnalyteSample.create(analyte_xml=analyte)
+            # Add the analyte to the well analyte mappings.
+            mappings.analytes.append(new_analyte)
+            # Update the analyte region dict
+            sample_documents.analyte_region_dict[
+                new_analyte.analyte_region
+            ] = new_analyte.analyte_name
+        sample_document = SampleDocument(
+            sample_type=sample_type,
+            sample_identifier=sample_id,
+            description=sample_description,
+            well_name=well_name,
+            well_analyte_mapping=mappings,
+            sample_dilution=sample_dilution,
+        )
+        return sample_document
+
+
+@dataclass
+class DeviceWellSettings:
+    # This data class is for all metadata needed in the <Wells> section, used in measurement and device control docs.
+    well_name: str
+    sample_volume_setting: int
+    detector_gain_setting: str
+    minimum_assay_bead_count_setting: int
+    well_total_events: int
+    acquisition_time: str
+
+    @staticmethod
+    def create(well_xml: ElementTree.Element) -> DeviceWellSettings:
+        well_name = get_well_name(well_xml.attrib)
+        well_acq_time = get_val_from_xml(well_xml, 2, "AcquisitionTime")
```

Comment:
See comment on get_val_from_xml about using names instead of numerical indexes. If possible, this would be much easier to follow / debug/ modify in the future. Also has the added benefit that you could probably use that for the error message :D

Reply:
Thank you for finding this, being able to not use numerical indexes is huge!